appsec/src/extension/commands_helpers.c

```diff
@@ -419,6 +420,23 @@ static void _command_process_redirect_parameters(mpack_node_t root)
 
     dd_set_redirect_code_and_location(status_code, location);
 }
+static void _command_process_stack_trace_parameters(mpack_node_t root)
+{
+    int expected_nodes = 1;
+    size_t count = mpack_node_map_count(root);
+    for (size_t i = 0; i < count && expected_nodes > 0; i++) {
+        mpack_node_t key = mpack_node_map_key_at(root, i);
+        mpack_node_t value = mpack_node_map_value_at(root, i);
+        if (dd_mpack_node_lstr_eq(key, "stack_id")) {
+            zend_string *id = NULL;
+            size_t id_len = mpack_node_strlen(value);
+            id = zend_string_init(mpack_node_str(value), id_len, 0);
+            report_backtrace(id);
+            zend_string_release(id);
+            --expected_nodes;
+        }
```

Comment:
```suggestion
    size_t count = mpack_node_map_count(root);
    for (size_t i = 0; i < count && expected_nodes > 0; i++) {
        mpack_node_t key = mpack_node_map_key_at(root, i);
        mpack_node_t value = mpack_node_map_value_at(root, i);
        if (dd_mpack_node_lstr_eq(key, "stack_id")) {
            zend_string *id = NULL;
            size_t id_len = mpack_node_strlen(value);
            id = zend_string_init(mpack_node_str(value), id_len, 0);
            report_backtrace(id);
            zend_string_release(id);
            break;
        }
```

I think it might be worth removing the extra logic to account for other parameters since there aren't any.

Reply:
Done